<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FillSettings</class>
 <widget class="QWidget" name="FillSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>243</width>
    <height>300</height>
   </rect>
  </property>
  <layout class="QGridLayout">
   <property name="leftMargin">
    <number>3</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="rightMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>3</number>
   </property>
   <property name="spacing">
    <number>3</number>
   </property>
   <item row="2" column="0" colspan="2">
    <widget class="KisSliderSpinBox" name="size">
     <property name="suffix">
      <string>px</string>
     </property>
     <property name="prefix">
      <string>Size Limit: </string>
     </property>
     <property name="minimum">
      <number>20</number>
     </property>
     <property name="maximum">
      <number>5000</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="blockUpdateSignalOnDrag" stdset="0">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="KisSliderSpinBox" name="tolerance">
     <property name="prefix">
      <string>Tolerance: </string>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="blockUpdateSignalOnDrag" stdset="0">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="11" column="1">
    <layout class="QHBoxLayout" name="modeLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="widgets::GroupedToolButton" name="areaContinuous">
       <property name="toolTip">
        <string>Fill continuous area</string>
       </property>
       <property name="statusTip">
        <string>Fill continuous area</string>
       </property>
       <property name="icon">
        <iconset theme="fill-color">
         <normaloff>theme:fill-color.svg</normaloff>theme:fill-color.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::GroupLeft</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="areaSimilar">
       <property name="toolTip">
        <string>Fill all similar colors</string>
       </property>
       <property name="statusTip">
        <string>Fill all similar colors</string>
       </property>
       <property name="icon">
        <iconset theme="color-picker">
         <normaloff>theme:color-picker.svg</normaloff>theme:color-picker.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::GroupCenter</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="areaSelection">
       <property name="toolTip">
        <string>Fill entire selection</string>
       </property>
       <property name="statusTip">
        <string>Fill entire selection</string>
       </property>
       <property name="icon">
        <iconset theme="select-rectangular">
         <normaloff>theme:select-rectangular.svg</normaloff>theme:select-rectangular.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::GroupRight</enum>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="modeSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Maximum</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>6</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="alphaPreserve">
       <property name="toolTip">
        <string notr="true">Preserve alpha</string>
       </property>
       <property name="statusTip">
        <string notr="true">Preserve alpha</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>theme:drawpile_alpha_unlocked.svg</normaloff>
         <normalon>theme:drawpile_alpha_locked.svg</normalon>theme:drawpile_alpha_unlocked.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::NotGrouped</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="blendModeCombo">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>24</width>
         <height>0</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="KisSliderSpinBox" name="opacity">
     <property name="suffix">
      <string>%</string>
     </property>
     <property name="prefix">
      <string>Opacity: </string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="blockUpdateSignalOnDrag" stdset="0">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="10" column="1">
    <layout class="QHBoxLayout" name="sourceLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="widgets::GroupedToolButton" name="sourceMerged">
       <property name="toolTip">
        <string>Merged image</string>
       </property>
       <property name="statusTip">
        <string>Merged image</string>
       </property>
       <property name="icon">
        <iconset theme="arrow-down-double">
         <normaloff>theme:arrow-down-double.svg</normaloff>theme:arrow-down-double.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::GroupLeft</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="sourceMergedWithoutBackground">
       <property name="toolTip">
        <string>Merged without background</string>
       </property>
       <property name="statusTip">
        <string>Merged without background</string>
       </property>
       <property name="icon">
        <iconset theme="arrow-down">
         <normaloff>theme:arrow-down.svg</normaloff>theme:arrow-down.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::GroupCenter</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="sourceLayer">
       <property name="toolTip">
        <string>Current layer</string>
       </property>
       <property name="statusTip">
        <string>Current layer</string>
       </property>
       <property name="icon">
        <iconset theme="layer-visible-on">
         <normaloff>theme:layer-visible-on.svg</normaloff>theme:layer-visible-on.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::GroupCenter</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="widgets::GroupedToolButton" name="sourceFillSource">
       <property name="toolTip">
        <string>Fill source layer</string>
       </property>
       <property name="statusTip">
        <string>Fill source layer</string>
       </property>
       <property name="icon">
        <iconset theme="tag">
         <normaloff>theme:tag.svg</normaloff>theme:tag.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="groupPosition">
        <enum>widgets::GroupedToolButton::GroupRight</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="sourceDummyCombo">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="14" column="0" colspan="2">
    <spacer name="bottomSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="10" column="0">
    <widget class="QLabel" name="sourceLabel">
     <property name="text">
      <string>Source:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="widgets::ExpandShrinkSpinner" name="expandShrink" native="true"/>
   </item>
   <item row="8" column="0" colspan="2">
    <widget class="KisSliderSpinBox" name="gap">
     <property name="suffix">
      <string>px</string>
     </property>
     <property name="prefix">
      <string>Close Gaps: </string>
     </property>
     <property name="maximum">
      <number>32</number>
     </property>
     <property name="blockUpdateSignalOnDrag" stdset="0">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="7" column="0" colspan="2">
    <widget class="KisSliderSpinBox" name="feather">
     <property name="suffix">
      <string>px</string>
     </property>
     <property name="prefix">
      <string>Feather: </string>
     </property>
     <property name="maximum">
      <number>40</number>
     </property>
     <property name="blockUpdateSignalOnDrag" stdset="0">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <widget class="QLabel" name="modeLabel">
     <property name="text">
      <string>Mode:</string>
     </property>
    </widget>
   </item>
   <item row="13" column="0" colspan="2">
    <layout class="QHBoxLayout" name="applyCancelLayout">
     <item>
      <widget class="QPushButton" name="applyButton">
       <property name="toolTip">
        <string>Apply the pending fill, making it visible for everyone</string>
       </property>
       <property name="statusTip">
        <string>Apply the pending fill, making it visible for everyone</string>
       </property>
       <property name="text">
        <string>Apply</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="toolTip">
        <string>Cancel the pending fill, without anyone else seeing its effects</string>
       </property>
       <property name="statusTip">
        <string>Cancel the pending fill, without anyone else seeing its effects</string>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KisSliderSpinBox</class>
   <extends>QSpinBox</extends>
   <header>desktop/widgets/kis_slider_spin_box.h</header>
  </customwidget>
  <customwidget>
   <class>widgets::GroupedToolButton</class>
   <extends>QToolButton</extends>
   <header>desktop/widgets/groupedtoolbutton.h</header>
  </customwidget>
  <customwidget>
   <class>widgets::ExpandShrinkSpinner</class>
   <extends>QWidget</extends>
   <header>desktop/widgets/expandshrinkspinner.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>size</tabstop>
  <tabstop>opacity</tabstop>
  <tabstop>tolerance</tabstop>
  <tabstop>feather</tabstop>
  <tabstop>gap</tabstop>
  <tabstop>sourceMerged</tabstop>
  <tabstop>sourceMergedWithoutBackground</tabstop>
  <tabstop>sourceLayer</tabstop>
  <tabstop>sourceFillSource</tabstop>
  <tabstop>sourceDummyCombo</tabstop>
  <tabstop>areaContinuous</tabstop>
  <tabstop>areaSimilar</tabstop>
  <tabstop>areaSelection</tabstop>
  <tabstop>alphaPreserve</tabstop>
  <tabstop>blendModeCombo</tabstop>
  <tabstop>applyButton</tabstop>
  <tabstop>cancelButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
